src/lib.rs

```diff
@@ -1,33 +1,43 @@
+mod libxenforeignmemory;
+
 extern crate xenforeignmemory_sys;
+
 use std::io::Error;
-use std::ptr::{null_mut};
-use std::os::raw::{c_int, c_ulong, c_void, c_uchar};
+use std::os::raw::{c_int, c_uchar, c_ulong, c_void};
+use std::ptr::null_mut;
 use std::slice;
 
+use libxenforeignmemory::LibXenForeignMemory;
+
 #[derive(Debug)]
 pub struct XenForeignMem {
     handle: *mut xenforeignmemory_sys::xenforeignmemory_handle,
+    libxenforeignmemory: LibXenForeignMemory,
 }
 
 impl XenForeignMem {
-
-    pub fn new() -> Result<Self,String> {
-        let fn_handle = unsafe {
-            xenforeignmemory_sys::xenforeignmemory_open(null_mut(), 0)
-        };
+    pub fn new() -> Result<Self, String> {
```

Comment:
I would create a custom error type instead of returning a string. There are libraries that take care of the boiler plate code for you. e.g. failure

Reply:
fixed